rfcs/libraries.md

```diff
@@ -0,0 +1,121 @@
+# Libraries
+
+## Summary
+
+We need a way to group together related code and provide easy ways to require them.
+
+## Motivation
+
+- Package management
+- Third party libraries
+- Simpler requires
+
+## Design
+
+### String Syntax
+
+Our require syntax should:
+
+- Work the same way in the Roblox engine and Luau language
+- Be resolvable at compile-time
+
+#### Absolute
+
+Modules can be required by their absolute path from the root of the filesystem.
+
+Requiring a file called `MyModule.luau` in `C:/MyLibrary`:
+```lua=
+local MyModule = require("C:/MyLibrary/MyModule")
+```
+
+#### Relative
+
+Modules can be required relative to the requiring files location in the filesystem.
+
+If we are trying to access a file called `MyModule.luau` in `C:/MyLibrary`:
+```lua=
+-- From C:/MyLibrary/SubDirectory/SubModule.luau
+local MyModule = require("../MyModule")
+ 
+-- From C:/MyOtherLibrary/MainModule.luau
+local MyModule = require("../MyLibrary/MyModule")
+```
+
+All relative paths will start with either `./`  or `../`  which denote the directory of the script or file, and parent directory respectively.
+
+#### Aliases
+
+Aliases can be used to bind an absolute or relative path to a convenient name that can be required directly. They are always prefixed with `$` to make it obvious they are not the same as a regular path.
+
+While the format is yet to be decided it will be a map that exists as part of a library and is defined in a format such as JSON.
+
+```json=
+{
+    "Roact": "C:/LuauModules/Roact-v1.4.2"
+}
+```
+
+Based on that map you would be able to require Roact directly:
+
+```lua=
+local Roact = require("$Roact")
```

Comment:
- Can aliases occur only at the beginning of a path?
- Can a string contain multiple aliases?

Reply:
- Aliases can only occur at the beginning of a path
- Multiple aliases are not supported

Reply:
Sounds good to me.